x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/DataNodeRequestSender.java

```diff
@@ -325,6 +342,11 @@ public void onSkip() {
                     onResponse(new DataNodeComputeResponse(DriverCompletionInfo.EMPTY, Map.of()));
                 }
             }
+
+            private void expectNoFailure(String message, Exception e) {
+                LOGGER.error(message, e);
+                assert false : new AssertionError(message, e);
+            }
```

Comment:
I believe the change to this file says "we expect no failures handling the response and if there is one log something big and loud and, if we're testing, crash loudly so we see if and fix it right away. If we're in production just be loud."

Reply:
Yes, that's correct. In production, if a request enters this state, we will fail it fast and log the error.